Mismatch credit handling

While compiling libfabric for i686 systems, I noticed there is a lot of warnings.
I'm working on fixing them (most are trivial issues).
But one is slightly trickier and I'm not familiar enough with ofi to know the besst way around


```
make CFLAGS="-m32  -Werror" V=1 
make  all-am
make[1]: Entering directory '/work1/nmorey/workspace/alternates/master/libfabric'
/bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.  -I./include -D_GNU_SOURCE -D__USE_XOPEN2K8 -DSYSCONFDIR=\"/usr/local/etc\" -DRDMADIR=\"@rdmadir@\" -DPROVDLDIR=\"/usr/local/lib/libfabric\" -I./prov/sockets/include -I./prov/sockets        -I./prov/hook/include -I./prov/hook/perf/include -I./prov/hook/hook_debug/include -I./prov/hook/hook_hmem/include -I./prov/hook/dmabuf_peer_mem/include  -Wall -m32  -Werror -MT prov/hook/src/src_libfabric_la-hook_domain.lo -MD -MP -MF prov/hook/src/.deps/src_libfabric_la-hook_domain.Tpo -c -o prov/hook/src/src_libfabric_la-hook_domain.lo `test -f 'prov/hook/src/hook_domain.c' || echo './'`prov/hook/src/hook_domain.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./include -D_GNU_SOURCE -D__USE_XOPEN2K8 -DSYSCONFDIR=\"/usr/local/etc\" -DRDMADIR=\"@rdmadir@\" -DPROVDLDIR=\"/usr/local/lib/libfabric\" -I./prov/sockets/include -I./prov/sockets -I./prov/hook/include -I./prov/hook/perf/include -I./prov/hook/hook_debug/include -I./prov/hook/hook_hmem/include -I./prov/hook/dmabuf_peer_mem/include -Wall -m32 -Werror -MT prov/hook/src/src_libfabric_la-hook_domain.lo -MD -MP -MF prov/hook/src/.deps/src_libfabric_la-hook_domain.Tpo -c prov/hook/src/hook_domain.c  -fPIC -DPIC -o prov/hook/src/.libs/src_libfabric_la-hook_domain.o
prov/hook/src/hook_domain.c: In function ‘hook_set_send_handler’:
prov/hook/src/hook_domain.c:124:12: error: passing argument 2 of ‘domain->base_ops_flow_ctrl->set_send_handler’ from incompatible pointer type [-Werror=incompatible-pointer-types]
            hook_credit_handler);
            ^~~~~~~~~~~~~~~~~~~
prov/hook/src/hook_domain.c:124:12: note: expected ‘ssize_t (*)(struct fid_ep *, uint64_t) {aka int (*)(struct fid_ep *, long long unsigned int)}’ but argument is of type ‘ssize_t (*)(struct fid_ep *, size_t) {aka int (*)(struct fid_ep *, unsigned int)}’
prov/hook/src/hook_domain.c: At top level:
prov/hook/src/hook_domain.c:150:17: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
  .add_credits = hook_add_credits,
                 ^~~~~~~~~~~~~~~~
prov/hook/src/hook_domain.c:150:17: note: (near initialization for ‘hook_ops_flow_ctrl.add_credits’)
prov/hook/src/hook_domain.c:152:22: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
  .set_send_handler = hook_set_send_handler,
                      ^~~~~~~~~~~~~~~~~~~~~
prov/hook/src/hook_domain.c:152:22: note: (near initialization for ‘hook_ops_flow_ctrl.set_send_handler’)
```

The issue is that `set_send_handler` in `ofi_util.h` expect the handler to use a uint64_t for credits, while the `base_credit_handler` from `ofi_hook.h` should use a size_t.
This works well on x86_64 systems, but may cause a mess on i686

Should the base_credit_handler be switched to uint64_t as well ?

--- a/include/ofi_hook.h
+++ b/include/ofi_hook.h
@@ -165,7 +165,7 @@ struct hook_domain {
 	struct fid_domain *hdomain;
 	struct hook_fabric *fabric;
 	struct ofi_ops_flow_ctrl *base_ops_flow_ctrl;
-	ssize_t (*base_credit_handler)(struct fid_ep *ep_fid, size_t credits);
+	ssize_t (*base_credit_handler)(struct fid_ep *ep_fid, uint64_t credits);
 };
 
 int hook_domain_init(struct fid_fabric *fabric, struct fi_info *info,

--- a/prov/hook/src/hook_domain.c
+++ b/prov/hook/src/hook_domain.c
@@ -102,7 +102,7 @@ static struct fi_ops_mr hook_mr_ops = {
 	.regattr = hook_mr_regattr,
 };
 
-static ssize_t hook_credit_handler(struct fid_ep *ep_fid, size_t credits)
+static ssize_t hook_credit_handler(struct fid_ep *ep_fid, uint64_t credits)
 {
 	/*
 	 * called from the base provider, ep_fid is the base ep, and
@@ -114,7 +114,7 @@ static ssize_t hook_credit_handler(struct fid_ep *ep_fid, size_t credits)
 }
 
 static void hook_set_send_handler(struct fid_domain *domain_fid,
-		ssize_t (*credit_handler)(struct fid_ep *ep, size_t credits))
+		ssize_t (*credit_handler)(struct fid_ep *ep, uint64_t credits))
 {
 	struct hook_domain *domain = container_of(domain_fid,
 						  struct hook_domain, domain);
@@ -131,7 +131,7 @@ static int hook_enable_ep_flow_ctrl(struct fid_ep *ep_fid, uint64_t threshold)
 	return ep->domain->base_ops_flow_ctrl->enable(ep->hep, threshold);
 }
 
-static void hook_add_credits(struct fid_ep *ep_fid, size_t credits)
+static void hook_add_credits(struct fid_ep *ep_fid, uint64_t credits)
 {
 	struct hook_ep *ep = container_of(ep_fid, struct hook_ep, ep);
 

--- a/prov/rxm/src/rxm_domain.c
+++ b/prov/rxm/src/rxm_domain.c
@@ -719,7 +719,7 @@ static struct fi_ops_mr rxm_domain_mr_thru_ops = {
 	.regattr = rxm_mr_regattr_thru,
 };
 
-static ssize_t rxm_send_credits(struct fid_ep *ep, size_t credits)
+static ssize_t rxm_send_credits(struct fid_ep *ep, uint64_t credits)
 {
 	struct rxm_conn *rxm_conn = ep->fid.context;
 	struct rxm_ep *rxm_ep = rxm_conn->ep;
@@ -772,12 +772,14 @@ defer:
 	return FI_SUCCESS;
 }
 
-static void rxm_no_add_credits(struct fid_ep *ep_fid, size_t credits)
-{ }
+static void rxm_no_add_credits(struct fid_ep *ep_fid, uint64_t credits)
+{
+}
 
 static void rxm_no_credit_handler(struct fid_domain *domain_fid,
-		ssize_t (*credit_handler)(struct fid_ep *ep, size_t credits))
-{ }
+		ssize_t (*credit_handler)(struct fid_ep *ep, uint64_t credits))
+{
+}
 
 static int rxm_no_enable_flow_ctrl(struct fid_ep *ep_fid, uint64_t threshold)
 {
@@ -858,7 +860,7 @@ static uint64_t rxm_get_coll_caps(struct fid_domain *domain)
 {
 	struct fi_collective_attr attr;
 	uint64_t mask = 0;
-	
+
 	attr.datatype = FI_INT8;
 	attr.datatype_attr.count = 1;
 	attr.datatype_attr.size = sizeof(int8_t);

--- a/prov/verbs/src/verbs_domain.c
+++ b/prov/verbs/src/verbs_domain.c
@@ -40,7 +40,7 @@
 
 
 static void vrb_set_credit_handler(struct fid_domain *domain_fid,
-		ssize_t (*credit_handler)(struct fid_ep *ep, size_t credits))
+		ssize_t (*credit_handler)(struct fid_ep *ep, uint64_t credits))
 {
 	struct vrb_domain *domain;
 

--- a/prov/verbs/src/verbs_ep.c
+++ b/prov/verbs/src/verbs_ep.c
@@ -38,7 +38,7 @@
 static struct fi_ops_msg vrb_srq_msg_ops;
 
 
-void vrb_add_credits(struct fid_ep *ep_fid, size_t credits)
+void vrb_add_credits(struct fid_ep *ep_fid, uint64_t credits)
 {
 	struct vrb_ep *ep;
 

